src/MergeContext.js

```diff
@@ -454,6 +454,270 @@ class BranchPosition
+
+            // treat an empty line as the end of input
+            if (line.trim().length === 0)
+                break;
+
+            if (/^\s/.test(line))
+                throw new Error(`a field cannot start with whitespace: ${line}`);
+
+            const pos = line.search(':');
+            if (pos < 0)
+                throw new Error(`a field without a name:value delimiter: '${line}'`);
+
+            const name = line.substring(0, pos);
+            const value = line.substring(pos+1);
+            if (this._remainingFields.some(el => el.name.toUpperCase() === name.toUpperCase() &&
+                        el.value.toUpperCase() === value.toUpperCase())) {
+                throw new Error(`duplicates are not allowed: ${line}`);
+            }
+            this._remainingFields.push({name: name, value: value, raw: line});
+        }
+    }
+
+    atEnd() { return this._remainingFields.length === 0; }
+
+    // zero or more paragraphs after the parsed fields and field terminator
+    remaining() { return this._lines.join('\n'); }
+}
+
+// computes future commit message from raw PR
+class CommitMessage
+{
+    constructor(rawPr, defaultAuthor) {
+
+        // the (required) commit message title
+        this._title = rawPr.title + ' (#' + rawPr.number + ')';
+
+        // PR description has three optional parts: [header]+[body]+[trailer]
+        // The parts are separated by empty lines.
+
+        // The header is dedicated to Anubis-recognized/consumed PR metadata.
+        // The commit message only retains the body and the trailer.
+
+        // PR description without the header and trailer parts
+        this._body = null;
+
+        // optional commit attributes recognized by GitHub, git, and/or Project
+        // these attributes are partially validated by Anubis
+        // stored in the original text format
+        this._trailer = null;
+
+        // GitHub-suggested commit author
+        // uses {name, email, date} format
+        assert(defaultAuthor);
+        this._defaultAuthor = defaultAuthor;
+
+        // optional commit author specified by 'Authored-by' header field
+        // uses {name, email} format
+        // overwrites this._defaultAuthor's name and email
+        this._customAuthor = null;
+
+        // We want to validate all trailers and only trailers for formatting
+        // errors. Unfortunately, there is no reliable way to detect trailer
+        // presence. Thus, we will miss some trailers (and merge some PRs with
+        // buggy trailers), and we will validate some non-trailers (and block
+        // some valid PRs). The current implementation considers excessive
+        // blocking to be the lesser of the two evils: We treat any last
+        // paragraph that starts with /\S+:/ as a trailer _unless_ it starts
+        // with one of these common expressions.
+        this.phonyTrailers = [
+            // common paragraph labels or introductory words
+            /^TODO:/i,
```

Comment:
I documented the current approach, but I believe it is not the best approach to use here. This hard-coded list of special non-trailer labels creates long-term code maintenance overheads, and we are going to annoy PR authors when their PRs are going to be incorrectly labeled/blocked. All that is not worth a slightly improved chance that we will notice a typo in a true trailer.

Given that in-scope typos ought to be very rare and virtually all PR descriptions are reviewed by humans anyway, I think we should chose to deal with the other evil (i.e., miss more typos) and use the opposite header/trailer detection algorithm: If paragraphLabel() contains a dash (`-`), then it is a header or trailer paragraph. Otherwise, it is not. Note that none of the existing phonyTrailers contain dashes so the whole array can be removed.

Reply:
> the other evil (i.e., miss more typos) and use the opposite header/trailer detection algorithm

Ok, let's do that.

> paragraphLabel() contains a dash (-), then it is a header or trailer paragraph

Also we should take into account that some legitimate one-word trailers could go without '-'. We cannot do this accurately, however, because there is no standard for Git message trailers (each platform such as GitHub supports its own set). For example, kernel developers use [this](https://git.wiki.kernel.org/index.php/CommitMessageConventions) list of trailers, and some of them lack '-'.

Reply:
> Also we should take into account that some legitimate one-word trailers could go without '-'.

Agreed. Anubis will mistreat last paragraphs that start with a one-word field as description body. Since the probability that mistreatment resulting in an actual problem (i.e. the rejection of the misidentified body due to a `Co-authored-by` or similar field that follows that one-word field) is very low and easily worked around without Anubis changes (by reordering trailer fields to place the `Co-authored-by` field first), I think we should accept that as the (far) lesser evil.

In the future, we can always add a list of recognized/honored trailer fields (and even a special marker to start the trailer) to tighten this up further if needed.